code/src/pages/MovieInfo.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect } from 'react'
+import { useParams, useHistory } from 'react-router-dom'
+
+import Loader from "react-loader-spinner"
+
+import { MOVIE_URL } from '../reusables/urls'
+
+const MovieInfo = () => {
+  const { id } = useParams()
+  const [movie, setMovie] = useState()
+  const [error, setError] = useState(false)
+  const [loading, setLoading] = useState(false)
+  const history = useHistory()
+
+  useEffect(() => {
+    setLoading(true)
+    fetch(MOVIE_URL(id))
+      .then(response => {
+        if (response.ok) {
+          return response.json()
+        } else throw new Error(response.statusText)
+      })
+      .then(receivedMovie => {
+        setMovie(receivedMovie)
+        setLoading(false)
+      })
+      .catch((error) => {
+        alert("Oops, something went wrong! You will be directed to the main page")
+        setError(true)
+      })
+  }, [id])
+  useEffect(() => {
+    if (error) {
+      history.push("/")
+    }
+  }, [error, history])
+
+  if (loading) {
+    return (
+      <div className="loading-text">
+        <Loader
+          className="loader"
+          type="ThreeDots"
+          color="#ffffff"
+          width={100}
```

Comment:
Great that you did a Loader! However, I would highly recommend do add a SetTimeout function in the useEffekt like this: 

`      setTimeout(()=>{
        setLoading(false)
      }, 1500)`

Or add a  timeout={3000}  in the `<Loader/> `. You can read more about it in the react-spinner-loader documentation.

 When I visited your website and clicked on Detail-Page I could see something go off very fast but not what it was, because the result from the fetch went really fast. So, with at setTimeout, you can control that so the loader shows for a second or more. Hope you understand my Swenglish!